File: C/杨辉三角.c
#include<stdio.h>
int main()
{
	int i, j, a[100][100];						//定义一个最大长度为100的二维数组 
	int n;
	printf("input n=:\n");
	scanf("%d", &n);

	for(i = 1; i < n; i++)
	{
		a[i][1] = 1;
		a[i][i] = 1;							//将二维数组的首行和对角线都赋值为1 
	}

	for(i = 3; i < n; i++)
	{
		for(j = 2; j < i; j++)					//由于是三角形，所以内层只循环i次 
		{
			a[i][j] = a[i-1][j-1] + a[i-1][j];	//对于内层非1的元素，都满足当前项 = 上一列对应的元素 + 上一列左边的元素 
		}
	}

	for(i = 1; i < n; i++)
	{
		for(j = 1; j <= i; j++)
		{ 
    		printf("%-2d ", a[i][j]);			//输出杨辉三角 
    	} 
    	printf("\n");
	}
	return 0; 
 }  

File: Java/杨辉三角.java
import java.util.*;
public class Ñî»ÔÈı½Ç {
	public static void main(String[] args) {
		System.out.print("input how many lines:");
		Scanner in = new Scanner(System.in);
		int lines = in.nextInt();
		lines = lines + 2;
		int[][] tr = new int[lines][lines];

		for (int i = 0; i < lines; i++) {
			tr[i][i] = 1;
			tr[i][0] = 1;
		}

		for (int i = 2; i < lines; i++) {
			for (int j = 1; j < i - 1; j++) {
				tr[i][j] = tr[i - 1][j - 1] + tr[i - 1][j];
			}
		}

		for (int i = 0; i < lines; i++) {
			for (int j = 0; j < i - 1; j++) {
				System.out.printf("%4d ", tr[i][j]);
			}
			System.out.print("\n");
		}
	}
}